### ISSUE TITLE
Shift+Click in add recipe interface crashes server

### ISSUE THREAD

**Issue by OffLuffy** [Jannyboy11/CustomRecipes#3]
In the process of adding a recipe for an elytra, I discovered that trying to use shift+click to move items into the interface causes the server to hang until the watchdog stops the server due to the thread becoming unresponsive. I tried this twice then manually moved items into the interface which then didn't crash the server. I'm not sure if this is a conflict with another plugin or something wrong with Paperclip.

This does not generate an error.

I'm using CustomRecipes v2.0.0-SNAPSHOT on Paperclip v1140 (1.12-R0.1-SNAPSHOT)

Here is the output from the watchdog when it detects the hang-up and shuts down (not that it looks terribly useful):
```
PID: 13 | Suspended: false | Native: false | State: WAITING
Stack:
java.lang.Object.wait(Native Method)
java.lang.Object.wait(Object.java:502)
java.util.TimerThread.mainLoop(Timer.java:526)
java.util.TimerThread.run(Timer.java:505)
------------------------------
Current Thread: Paper Async Task Handler Thread - 0
PID: 12 | Suspended: false | Native: false | State: WAITING
Stack:
sun.misc.Unsafe.park(Native Method)
java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
java.lang.Thread.run(Thread.java:745)
------------------------------
Current Thread: Signal Dispatcher
PID: 5 | Suspended: false | Native: false | State: RUNNABLE
Stack:
------------------------------
Current Thread: Finalizer
PID: 3 | Suspended: false | Native: false | State: WAITING
Stack:
java.lang.Object.wait(Native Method)
java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)
------------------------------
Current Thread: Reference Handler
PID: 2 | Suspended: false | Native: false | State: WAITING
Stack:
java.lang.Object.wait(Native Method)
java.lang.Object.wait(Object.java:502)
java.lang.ref.Reference.tryHandlePending(Reference.java:191)
java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)
------------------------------
```

**Comment by Jannyboy11** [comment]
Having the same issue with the latest 'regular' Spigot without other plugins. I think it's weird though, I'm not using any locking in my code, and clearly the output shows multiple threads are waiting to be unlocked. I don't want to say it's a bug in minecraft/spigot just yet, but looks like it very much. Needs further investigation.

**Comment by Jannyboy11** [comment]
My logs:

[10:54:20] [Server thread/INFO]: [CustomRecipes] Enabling CustomRecipes v2.0.0-SNAPSHOT
[10:54:20] [Server thread/INFO]: Server permissions file permissions.yml is empty, ignoring it
[10:54:20] [Server thread/INFO]: Done (4,040s)! For help, type "help" or "?"
[10:54:24] [User Authenticator #1/INFO]: UUID of player Jannyboy11 is 7940653f-a153-47e1-a184-c30a5075a2f5
[10:54:24] [Server thread/INFO]: Jannyboy11[/127.0.0.1:52073] logged in with entity id 56 at ([world]-133.92954795834035, 68.0, 197.03773439548974)
[10:54:34] [Server thread/INFO]: Jannyboy11 issued server command: /addrecipe shaped fubaryolo
[10:55:45] [Spigot Watchdog Thread/ERROR]: The server has stopped responding!
[10:55:45] [Spigot Watchdog Thread/ERROR]: Please report this to http://www.spigotmc.org/
[10:55:45] [Spigot Watchdog Thread/ERROR]: Be sure to include ALL relevant console errors and Minecraft crash reports
[10:55:45] [Spigot Watchdog Thread/ERROR]: Spigot version: git-Spigot-7228328-d603539 (MC: 1.12)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Spigot!):
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Server thread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 17 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.Item.getItemOf(SourceFile:108)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.ItemStack.isEmpty(ItemStack.java:130)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.Container.a(Container.java:251)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1885)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:33)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:10)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.concurrent.FutureTask.run(FutureTask.java:266)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:747)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Entire Thread Dump:
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: threadDeathWatcher-2-1
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 27 | Suspended: false | Native: false | State: TIMED_WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.sleep(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.util.ThreadDeathWatcher$Watcher.run(ThreadDeathWatcher.java:152)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Netty Server IO #1
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 26 | Suspended: false | Native: true | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Thread is waiting on monitor(s):
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(WindowsSelectorImpl.java:296)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(WindowsSelectorImpl.java:278)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl.doSelect(WindowsSelectorImpl.java:159)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:732)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:388)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: File IO Thread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 25 | Suspended: false | Native: false | State: TIMED_WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.sleep(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.FileIOThread.c(SourceFile:51)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.FileIOThread.run(SourceFile:30)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Netty Server IO #0
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 24 | Suspended: false | Native: true | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Thread is waiting on monitor(s):
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(WindowsSelectorImpl.java:296)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(WindowsSelectorImpl.java:278)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.WindowsSelectorImpl.doSelect(WindowsSelectorImpl.java:159)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:732)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:388)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Spigot Metrics Thread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 23 | Suspended: false | Native: false | State: TIMED_WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.TimerThread.mainLoop(Timer.java:552)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.TimerThread.run(Timer.java:505)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Spigot Watchdog Thread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 22 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.management.ThreadImpl.dumpThreads0(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:454)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.spigotmc.WatchdogThread.run(WatchdogThread.java:76)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Server console handler
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 20 | Suspended: false | Native: true | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Thread is waiting on monitor(s):
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:org.bukkit.craftbukkit.libs.jline.internal.NonBlockingInputStream.read(NonBlockingInputStream.java:169)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:org.bukkit.craftbukkit.libs.jline.internal.InputStreamReader.read(InputStreamReader.java:261)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:org.bukkit.craftbukkit.libs.jline.internal.InputStreamReader.read(InputStreamReader.java:198)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.fusesource.jansi.internal.Kernel32.ReadConsoleInputW(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.fusesource.jansi.internal.Kernel32.readConsoleInputHelper(Kernel32.java:761)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.fusesource.jansi.internal.Kernel32.readConsoleKeyInput(Kernel32.java:794)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.fusesource.jansi.internal.WindowsSupport.readConsoleInput(WindowsSupport.java:97)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.WindowsTerminal.readConsoleInput(WindowsTerminal.java:215)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.WindowsTerminal.access$000(WindowsTerminal.java:55)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.WindowsTerminal$1.read(WindowsTerminal.java:157)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.NonBlockingInputStream.read(NonBlockingInputStream.java:169)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.NonBlockingInputStream.read(NonBlockingInputStream.java:137)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.NonBlockingInputStream.read(NonBlockingInputStream.java:246)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.InputStreamReader.read(InputStreamReader.java:261)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.InputStreamReader.read(InputStreamReader.java:198)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.console.ConsoleReader.readCharacter(ConsoleReader.java:2145)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.console.ConsoleReader.readLine(ConsoleReader.java:2349)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer$2.run(DedicatedServer.java:85)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Thread-4
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 21 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Thread is waiting on monitor(s):
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:java.io.PrintStream.write(PrintStream.java:451)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:java.io.PrintStream.write(PrintStream.java:480)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		Locked on:java.io.PrintStream.write(PrintStream.java:480)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.FileOutputStream.writeBytes(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.FileOutputStream.write(FileOutputStream.java:326)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.PrintStream.write(PrintStream.java:451)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.fusesource.jansi.AnsiOutputStream.write(AnsiOutputStream.java:82)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.FilterOutputStream.write(FilterOutputStream.java:125)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.PrintStream.write(PrintStream.java:480)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.PrintStream.write(PrintStream.java:480)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.io.FilterOutputStream.write(FilterOutputStream.java:97)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.util.TerminalConsoleWriterThread.run(TerminalConsoleWriterThread.java:36)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: DestroyJavaVM
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 19 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Server thread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 17 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.Container.a(Container.java:251)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1885)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:33)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:10)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.concurrent.FutureTask.run(FutureTask.java:266)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:747)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Server Infinisleeper
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 18 | Suspended: false | Native: false | State: TIMED_WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.sleep(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer$1.run(DedicatedServer.java:60)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: NonBlockingInputStreamThread
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 15 | Suspended: false | Native: false | State: WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.libs.jline.internal.NonBlockingInputStream.run(NonBlockingInputStream.java:278)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:745)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Snooper Timer
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 13 | Suspended: false | Native: false | State: TIMED_WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.TimerThread.mainLoop(Timer.java:552)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.util.TimerThread.run(Timer.java:505)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Attach Listener
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 5 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Signal Dispatcher
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 4 | Suspended: false | Native: false | State: RUNNABLE
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Finalizer
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 3 | Suspended: false | Native: false | State: WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/ERROR]: Current Thread: Reference Handler
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	PID: 2 | Suspended: false | Native: false | State: WAITING
[10:55:45] [Spigot Watchdog Thread/ERROR]: 	Stack:
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Native Method)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.Object.wait(Object.java:502)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.ref.Reference.tryHandlePending(Reference.java:191)
[10:55:45] [Spigot Watchdog Thread/ERROR]: 		java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)
[10:55:45] [Spigot Watchdog Thread/ERROR]: ------------------------------
[10:55:45] [Spigot Watchdog Thread/INFO]: Startup script './start.sh' does not exist! Stopping server.
[10:55:45] [Spigot Watchdog Thread/INFO]: Stopping server
[10:55:45] [Spigot Watchdog Thread/INFO]: [CustomRecipes] Disabling CustomRecipes v2.0.0-SNAPSHOT
[10:55:45] [Spigot Watchdog Thread/INFO]: Saving players
[10:55:45] [Spigot Watchdog Thread/INFO]: Jannyboy11 lost connection: Server closed
[10:55:45] [Spigot Watchdog Thread/INFO]: Jannyboy11 left the game
[10:55:45] [Spigot Watchdog Thread/INFO]: Saving worlds
[10:55:45] [Spigot Watchdog Thread/INFO]: Saving chunks for level 'world'/overworld
[10:55:46] [Spigot Watchdog Thread/INFO]: Saving chunks for level 'world_nether'/the_nether
[10:55:46] [Spigot Watchdog Thread/INFO]: Saving chunks for level 'world_the_end'/the_end

**Comment by OffLuffy** [comment]
It seems this also happens with some other inventory interactions, such as click+drag.

Most obvious guess is that the inventory interact event is entering into some sort of infinite loop that locks up the main thread.

**Comment by Jannyboy11** [comment]
Do you mean that an eventhandler method fires a new event that results in a new invocation of the eventhandler? I would expect a StackOverFlowError in that case.

**Comment by Jannyboy11** [comment]
Sorry for the inactivity on this issue. I did some testing and eventually [reported this issue on the SpigotMC JIRA](https://hub.spigotmc.org/jira/browse/SPIGOT-3500) cause I couldn't fix it.

**Comment by Jannyboy11** [comment]
Confirmed spigot bug. I can't really fix it in this plugin, so I'm closing the issue. Feel free to comment still if you like.

**Comment by Jannyboy11** [comment]
Fixed as of [79e55b6dcf6](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/79e55b6dcf6448d33465b936ebdb2cc8a3cf2383#src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java)
